src/App.tsx

```diff
@@ -1,27 +1,72 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[] | void>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const getAllGoods = async () => {
+    try {
+      const allGoods = await goodsAPI.getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      setGoods(allGoods);
+    } catch (error) {
+      throw new Error('error');
+    }
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const get5FirstGoods = async () => {
+    try {
+      const filteredGoods = await goodsAPI.get5First();
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      setGoods(filteredGoods);
+    } catch (error) {
+      throw new Error('error');
+    }
+  };
+
+  const getRedGoods = async () => {
+    try {
+      const filteredGoods = await goodsAPI.getRed();
+
+      setGoods(filteredGoods);
+    } catch (error) {
+      throw new Error('error');
+    }
+  };
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={getAllGoods}
+      >
+        Load all goods
+      </button>
+
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={get5FirstGoods}
+      >
+        Load 5 first goods
+      </button>
+
+      <button
+        type="button"
+        data-cy="red-button"
+        onClick={getRedGoods}
+      >
+        Load red goods
+      </button>
+
+      {goods && (<GoodsList goods={goods} />)}
```

Comment:
```suggestion
      {!!goods.length && (<GoodsList goods={goods} />)}
```
or
```suggestion
   <GoodsList goods={goods} />
```

Reply:
goods always will be true.
check its length or not check nothing. 
empty array will not be rendered